sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/workflow/WorkflowRouter.scala

```diff
@@ -81,8 +82,11 @@ abstract class WorkflowRouter[S, W <: Workflow[S]](protected val workflow: W) {
 
   /** INTERNAL API */
   // "public" api against the impl/testkit
-  def _internalSetInitState(s: Any): Unit = {
-    state = Some(s.asInstanceOf[S])
+  def _internalSetInitState(s: Any, finished: Boolean): Unit = {
```

Comment:
ok, this will work also for recovering a finished workflow, since then the flag is false and then true